contracts/DCNTCrescendo.sol

```diff
@@ -129,8 +133,17 @@ contract DCNTCrescendo is
     return ((bps - takeRateBPS) * _currentPrice[id]) / bps;
   }
 
-  function buy(uint256 id) external payable {
-    require(saleIsActive, "Sale must be active to buy");
+  modifier salesAreActive {
+    require(block.timestamp >= saleStart, "Sales are not active yet.");
```

Comment:
\>= / <= costs more gas than > < recommend updating for gas optimization.

https://nftchance.medium.com/the-gas-efficient-way-of-building-and-launching-an-erc721-nft-project-for-2022-b3b1dac5f2e1